src/main/java/de/embl/cba/mobie/bookmark/BookmarksManager.java

```diff
@@ -75,6 +91,131 @@ public void adaptViewerTransform( Bookmark bookmark )
 		}
 	}
 
+	public void loadAdditionalBookmarks() {
+			Map<String, Bookmark> additionalBookmarks = bookmarksJsonParser.selectAndLoadBookmarks();
+			if (additionalBookmarks != null) {
+				nameToBookmark.putAll(additionalBookmarks);
+				bookmarkDropDown.removeAllItems();
+				for (String bookmarkName : nameToBookmark.keySet()) {
+					bookmarkDropDown.addItem(bookmarkName);
+				}
+			}
+	}
+
+	public void saveCurrentSettingsAsBookmark () {
+		ArrayList<String> bookmarkNameAndLocation = bookmarkSaveDialog();
+		Bookmark currentBookmark = getBookmarkFromCurrentSettings(bookmarkNameAndLocation.get(0));
+		ArrayList<Bookmark> bookmarks = new ArrayList<>();
+		bookmarks.add(currentBookmark);
+
+		if (bookmarkNameAndLocation.get(1).equals(FileUtils.PROJECT) &&
+				bookmarksJsonParser.getDatasetLocation().contains( "raw.githubusercontent" )) {
+			bookmarksJsonParser.saveBookmarksToGithub(bookmarks);
+		} else {
+			try {
+				bookmarksJsonParser.saveBookmarksToFile(bookmarks, bookmarkNameAndLocation.get(1));
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+		}
+	}
+
+	private ArrayList<String> bookmarkSaveDialog () {
+		String fileLocation = null;
+		String bookmarkName = null;
+		final GenericDialog gd = new GenericDialog( "Choose save location" );
+		gd.addStringField("Bookmark Name", "name");
+		gd.addChoice( "Save to", new String[]{FileUtils.PROJECT, FileUtils.FILE_SYSTEM }, FileUtils.PROJECT );
+		gd.showDialog();
+		if ( gd.wasCanceled() ) return null;
+		bookmarkName = gd.getNextString();
+		fileLocation = gd.getNextChoice();
+
+		ArrayList<String> bookmarkNameandLocation = new ArrayList<>();
+		bookmarkNameandLocation.add(bookmarkName);
+		bookmarkNameandLocation.add(fileLocation);
+		return bookmarkNameandLocation;
+	}
+
+	public Bookmark getBookmarkFromCurrentSettings( String bookmarkName) {
```

Comment:
My personal convention is (and there is no general convention):

- If a method `creates` something new I call it `createXYZ( )`
- If a method just returns something that exists already I call it `getXYZ( )`

In this case I would thus vote for calling it:

`createBookmarkFromCurrentSettings( String newBookmarkName )`

Otherwise I find it confusing, the way it is right now it sounds more like it tries to fetch a bookmark with the name `bookmarkName` from an already existing list of bookmarks.